modules/claims_api/app/sidekiq/claims_api/va_notify_declined_job.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class VANotifyDeclinedJob < ClaimsApi::ServiceBase
+    LOG_TAG = 'va_notify_declined_job'
+
+    def perform(encrypted_ptcpnt_id:, encrypted_first_name:, poa_code:)
+      lockbox = Lockbox.new(key: Settings.lockbox.master_key)
+      ptcpnt_id = lockbox.decrypt(encrypted_ptcpnt_id)
+      first_name = lockbox.decrypt(encrypted_first_name)
+      poa = find_poa(poa_code:)
+
+      if poa.blank?
+        raise ClaimsApi::Common::Exceptions::Lighthouse::ResourceNotFound.new(
+          detail: "Could not find Power of Attorney with POA code: #{poa_code}"
+        )
+      end
+
+      res = send_declined_notification(ptcpnt_id:, first_name:, poa:)
+
+      ClaimsApi::VANotifyFollowUpJob.perform_async(res.id) if res.present?
+    rescue => e
+      msg = "VA Notify email notification failed to send with error #{e}"
+      slack_client = SlackNotify::Client.new(webhook_url: Settings.claims_api.slack.webhook_url,
+                                             channel: '#api-benefits-claims-alerts',
+                                             username: 'Failed ClaimsApi::VANotifyDeclinedJob')
+
+      slack_client.notify(msg)
+      ClaimsApi::Logger.log(LOG_TAG, detail: msg)
+
+      raise e
+    end
+
+    private
+
+    def find_poa(poa_code:)
+      ClaimsApi::PowerOfAttorney.find do |poa|
+        poa.form_data.dig('serviceOrganization', 'poaCode') == poa_code ||
+          poa.form_data.dig('representative', 'poaCode') == poa_code
+      end
```

Comment:
Is there a better way to get the `PowerOfAttorney` record without a full table scan?

Also, I couldnʼt find any production records where `poa.form_data['representative'].present?` was true. Are we sure [form_data['representative']](https://github.com/department-of-veterans-affairs/vets-api/blob/a79d93dd9e7d48509b0f5edb423d83d9916825ec/modules/claims_api/app/sidekiq/claims_api/service_base.rb#L224-L225) is valid?

Reply:
there's discussion of adding a representative ID to the request due to the need for it to be provided in the auto establishment process. That being said, we could use that to locate a Veteran Representative. We also want to do similar lookups to the `poa_verfication.rb` in respect to this, so instead of looking at a `ClaimsApi::PowerOfAttorney` which is a DB entry created when kicking off the auto establishment, we'll want to look up a `::Veteran::Service::Representative`.. so something along the lines of `::Veteran::Service::Representative.find_by(representative_id: 'blah')`

Reply:
Interesting, that would simplify this logic. We can add representative_id to the request params and do the straightforward lookup here.